packages/gitbook/src/app/(site)/fetch.ts

```diff
@@ -33,19 +33,28 @@ export async function fetchContentData() {
                 organizationId: content.organizationId,
                 siteId: content.siteId,
                 siteShareKey: content.siteShareKey,
+                siteSpaceId: content.siteSpaceId,
             }),
         ]);
 
     const site = siteStructure.site;
-    const spaces = siteStructure.spaces;
+
+    const sections = siteStructure.sections;
+    const section = siteStructure.sections?.find((section) => section.id === content.siteSectionId);
+
+    const spaces =
+        siteStructure.spaces ?? (section ? parseSpacesFromSiteSpaces(section.siteSpaces) : []);
+
     // we grab the space attached to the parent as it contains overriden customizations
-    const spaceRelativeToParent = spaces.find((space) => space.id === content.spaceId);
+    const spaceRelativeToParent = spaces?.find((space) => space.id === content.spaceId);
 
     return {
         content,
         contentTarget,
         space: spaceRelativeToParent ?? space,
         pages,
+        sections,
+        section,
```

Comment:
On that same note the return typing should also indicate that `section` would always exist if site has sections. We could maybe refactor this to be

```js
return {
  ...,
 sections: sections ? { list: sections, section } : undefined
}
```